### PR TITLE
Add Crust Network Support (No Chain Events)

### PR DESCRIPTION
**Pull request by zakhap**
## Description

- [x] Migration 
- [x] Add to app.ts, types, etc. 
- [x] Add generic substrate controller
- [x] Add CrustSpec

## Motivation and Context

- Adding new Polkadot networks!

## How has this been tested?

- Not yet. 

## Have proper tags been added (for bug, enhancement, breaking change)?
- [x] yes

## Does this PR affect any server routes?
- [x] no

### CODE REVIEW

File: server/migrations/20210512232604-add-crust.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+const SubstrateEventKinds = {
```
Comment (jnaviask) [inline server/migrations/20210512232604-add-crust.js:3]: These are no longer necessary.

File: server/migrations/20210512232604-add-crust.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  CollectiveMemberExecuted: 'collective-member-executed',
+};
+
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.sequelize.transaction(async (t) => {
+      await queryInterface.bulkInsert('Chains', [{
+        id: 'crust',
+        symbol: 'CRUST',
+        name: 'Crust',
```
Comment (jnaviask) [inline server/migrations/20210512232604-add-crust.js:48]: Needs a `base` field.

File: server/migrations/20210512232604-add-crust.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        chat: 'https://discord.com/invite/Jbw2PAUSCR',
+        github: 'https://github.com/crustio',
+      }], { transaction: t });
+
+      await queryInterface.bulkInsert('ChainNodes', [{
+        chain: 'crust',
+        url: 'wss://api.crust.network/',
+      }], { transaction: t });
+
+      const buildObject = (event_name, chain) => ({
```
Comment (jnaviask) [inline server/migrations/20210512232604-add-crust.js:65]: You can get rid of all this too, the chain event creation stuff.

File: shared/adapters/chain/crust/spec.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
+const CrustSpec = {
```
Comment (jnaviask) [inline shared/adapters/chain/crust/spec.ts:1]: Need to add the types key so it's like `{ types: { ...CrustSpec } }`.

**Comment by zakhap** [comment]
Seems like most things are ok, but we're receiving malformed data from the official crust node.
Checked if working:
- [x] Chain-Initialization
- [x] Councillors
- [ ] Validators
- [ ] Referenda/Proposals
- [ ] Bounty/Treasury

The pages that aren't working throw this error: 
`storage system.account: entry 0:: createType(AccountInfo):: {"nonce":"Index","consumers":"RefCount","providers":"RefCount","sufficients":"RefCount","data":"AccountData"}:: Decoded input doesn't match input, received 0x000000000000000058f8f8bd4890020000000000000000000000000000000000…0000000000000000000000000000000000000000000000000000000000000000 (72 bytes), created 0x000000000000000058f8f8bd4890020000000000000000000000000000000000…0000000000000000000000000000000000000000000000000000000000000000 (80 bytes)`

**Comment by zakhap** [comment]
Update on debugging the error loading some of the substrate governance modules:

- It seems that the core of the issue is being thrown from `@polkadot/rpc-core`
- After the error is thrown above, it seems to not throw any other similar errors when accessing other known 'malformed data' pages, while leaving the spinning loading wheel indefinitely.

**Comment by zakhap** [comment]
Just fixed the last issue with the UI, when the validators don't have a commission (doesn't seem like any of them do on the crust network), it just doesn't render an invalid stat.

**Pull request merged**